src/homepageExperience/components/steps/Finish.tsx

```diff
@@ -82,22 +87,33 @@ export const Finish = (props: OwnProps) => {
         Curious to learn more? Try these next steps!
       </p>
       <FlexBox margin={ComponentSize.Medium} alignItems={AlignItems.Stretch}>
-        <ResourceCard className="homepage-wizard-next-steps">
-          <SafeBlankLink href="https://github.com/InfluxCommunity/sample-flask/blob/main/app.py">
-            <h4>{CodeTerminalIcon}Sample App</h4>
-          </SafeBlankLink>
-          <p>
-            Play around with our template code of sample app to streamline your
-            own data into InfluxData.
-          </p>
-        </ResourceCard>
-        <ResourceCard className="homepage-wizard-next-steps">
+        {showSampleApp && (
+          <ResourceCard
+            className="homepage-wizard-next-steps"
+            onClick={() => handleNextStepEvent('sampleApp')}
+          >
+            <SafeBlankLink href="https://github.com/InfluxCommunity/sample-flask/blob/main/app.py">
+              <h4>{CodeTerminalIcon}Sample App</h4>
+            </SafeBlankLink>
+            <p>
+              Play around with our template code of sample app to streamline
+              your own data into InfluxData.
+            </p>
+          </ResourceCard>
+        )}
+        <ResourceCard
+          className="homepage-wizard-next-steps"
+          onClick={() => handleNextStepEvent('keyConcepts')}
```

Comment:
this anon function seems to be the only way to handle 3 different cases. open to a better option tho!